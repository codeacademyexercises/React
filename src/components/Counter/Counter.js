import React, { Component } from 'react';
import Button from '../Button';
// import logo from './logo.svg';
// import './App.css';

class Counter extends Component {
    state = {
        count: this.props.initialValue,
    };
    increment = ()=> this.setState({count: this.state.count+1});
    decrement = ()=> this.setState({count: this.state.count-1});    
  render() {
      console.log(this.state.count);
    return (
      <div>
          <div>Counter :{this.state.count}</div>
          <Button name="Add" onClick={this.increment}></Button>
          <Button name="Minus" onClick={this.decrement}></Button>
      </div>
    );
  }
}

export default Counter;
