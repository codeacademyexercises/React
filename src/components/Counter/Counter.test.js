import React from 'react';
// import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
// import {shallow} from 'enzyme';
import Counter from './Counter.js';
import {shallow} from 'enzyme';

describe('Button ',()=>{
    it('renderer should display button',()=>{
        const tree = renderer.create(<Counter/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should trigger onClick on clicking the button',()=>{
        const wrapper = shallow(<Counter name="testing" onClick={jest.fn()}/>);
        wrapper.find('button').simulate('click');
        expect(wrapper.instance().props.onClick).toHaveBeenCalled();
    });
});