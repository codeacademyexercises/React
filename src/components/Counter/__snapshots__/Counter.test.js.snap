// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  renderer should display button 1`] = `
<div>
  <div>
    Counter :
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
  </div>
  <div>
    <button
      onClick={[Function]}
      type="button"
    >
      Minus
    </button>
  </div>
</div>
`;
